services/user.js

```diff
@@ -40,13 +40,43 @@ class User {
     async grantToken(params, context){
         isLoggedInOrThrow(context);
         isValidUserStatus(context?.userInfo?.userStatus);
-        let accessToken = createToken(context?.userInfo, config.token_secret, config.tokenTimeout)
+        let accessToken = createToken(context?.userInfo, config.token_secret, config.tokenTimeout);
+        const aUser = await this.linkTokentoUser(context, accessToken);
         return {
             tokens: [accessToken],
             message: "This token can only be viewed once and will be lost if it is not saved by the user"
         }
     }
 
+    async linkTokentoUser(context, accessToken){
+        const sessionCurrentTime = getCurrentTime();
+        const user = await this.userCollection.find(context.userInfo._id);
```

Comment:
This user variable isn't needed until after the update. Since the update can potentially fail and the user won't be needed, it is best practice to not retrieve this until it is needed.

This might not be needed at all if we don't need a return value